includes/razorpay-subscriptions.php

```diff
@@ -0,0 +1,192 @@
+<?php
+
+require_once __DIR__.'/../razorpay-sdk/Razorpay.php';
+
+use Razorpay\Api\Api;
+
+class RZP_Subscriptions
+{
+    public function __construct($keyId, $keySecret)
+    {
+        $this->api = new Api($keyId, $keySecret);
+    }
+
+    public function createSubscription($orderId)
+    {
+        global $woocommerce;
+
+        $subscriptionData = $this->getSubscriptionCreateData($orderId);
+
+        $subscription = $this->api->subscription->create($subscriptionData);
+
+        // Setting the subscription id as the session variable
+        $sessionKey = $this->getSubscriptionSessionKey($orderId);
+        $woocommerce->session->set($sessionKey, $subscription['id']);
+
+        return $subscription['id'];
+    }
+
+    protected function getProduct($order)
+    {
+        $products = $order->get_items();
+
+        // Technically, subscriptions work only if there's one array in the cart
+        if (sizeof($products) > 1)
+        {
+            throw new Exception('Currently Razorpay does not support more than
+                                one product in the cart if one of products
+                                is a subscription.');
+        }
+
+        return array_values($products)[0];
+    }
+
+    protected function getSubscriptionCreateData($orderId)
+    {
+        $order = new WC_Order($orderId);
+
+        $product = $this->getProduct($order);
+
+        $planId = $this->getProductPlanId($product);
+
+        $customerId = $this->getCustomerId($order);
+
+        $length = (int) WC_Subscriptions_Product::get_length($product['product_id']);
```

Comment:
@ccrims0n We discussed to leave it as int, and either way does not make a difference.